src/ImageSharp/Formats/Png/PngDecoderCore.cs

```diff
@@ -876,12 +883,151 @@ private void ReadTextChunk(ImageMetadata metadata, ReadOnlySpan<byte> data)
+            }
+
+            byte compressionFlag = data[zeroIndexKeyword + 1];
+            if (!(compressionFlag == 0 || compressionFlag == 1))
+            {
+                return;
+            }
+
+            byte compressionMethod = data[zeroIndexKeyword + 2];
+            if (compressionMethod != 0)
+            {
+                // Only compression method 0 is supported (zlib datastream with deflate compression).
+                return;
+            }
+
+            int langStartIdx = zeroIndexKeyword + 3;
+            int languageLength = data.Slice(langStartIdx).IndexOf((byte)0);
+            if (languageLength < 0)
+            {
+                return;
+            }
+
+            string language = this.latinEncoding.GetString(data.Slice(langStartIdx, languageLength));
+
+            int translatedKeywordStartIdx = langStartIdx + languageLength + 1;
+            int translatedKeywordLength = data.Slice(translatedKeywordStartIdx).IndexOf((byte)0);
+            string translatedKeyword = this.latinEncoding.GetString(data.Slice(translatedKeywordStartIdx, translatedKeywordLength));
+
+            ReadOnlySpan<byte> keywordBytes = data.Slice(0, zeroIndexKeyword);
+            if (this.TryReadTextKeyword(keywordBytes, out string name))
+            {
+                return;
+            }
+
+            int dataStartIdx = translatedKeywordStartIdx + translatedKeywordLength + 1;
+            if (compressionFlag == 1)
+            {
+                ReadOnlySpan<byte> compressedData = data.Slice(dataStartIdx);
+                metadata.Properties.Add(new ImageProperty(name, this.UncompressTextData(compressedData, Encoding.UTF8)));
+            }
+            else
+            {
+                string value = Encoding.UTF8.GetString(data.Slice(dataStartIdx));
+                metadata.Properties.Add(new ImageProperty(name, value));
+            }
+        }
+
+        /// <summary>
+        /// Decompresses a zlib text.
+        /// </summary>
+        /// <param name="compressedData">Compressed text data bytes.</param>
+        /// <param name="encoding">The string encoding to use.</param>
+        /// <returns>A string.</returns>
+        private string UncompressTextData(ReadOnlySpan<byte> compressedData, Encoding encoding)
+        {
+            using (var memoryStream = new MemoryStream(compressedData.ToArray()))
+            using (var inflateStream = new ZlibInflateStream(memoryStream, () => 0))
+            {
+                inflateStream.AllocateNewBytes(compressedData.Length);
+                var uncompressedBytes = new List<byte>();
+                int bytesRead = inflateStream.CompressedStream.Read(this.buffer, 0, this.buffer.Length);
+                while (bytesRead != 0)
+                {
+                    uncompressedBytes.AddRange(this.buffer.AsSpan().Slice(0, bytesRead).ToArray());
+                    bytesRead = inflateStream.CompressedStream.Read(this.buffer, 0, this.buffer.Length);
```

Comment:
@JimBobSquarePants: i was a bit unsure about this part. Im re-using here the buffer from the PngDecoderCore to uncompress, but its only 4 bytes long. Does it make sense here for performance reasons to allocate a larger buffer? Or is it ok like it is?

Reply:
Oh yeah, I see what you mean. I'm not too concerned as I don't think these tags are commonly used. Maybe we can just add a note for now.